sopel/modules/units.py

```diff
@@ -44,13 +52,13 @@ def k_to_c(temp):
 def temperature(bot, trigger):
     """Convert temperatures"""
     try:
-        source = find_temp.match(trigger.group(2)).groups()
-    except (AttributeError, TypeError):
+        source = _extract_source(find_temp, trigger)
+    except Exception:
```

Comment:
Part of the reason I made this change is because the `AttributeError` was doing work that should have been done by checking the result of the `re` call. I'm happy to revert, but I generalized because I don't see what exceptions it would be important to let propagate here, and _"if something goes wrong, the bot says so"_ seems to be the intended behavior.

Reply:
I, for one, like to differentiate between "something went wrong, but we expect that sometimes" and "something went wrong, and it means some other thing broke/changed". Listing the specific exceptions that can be expected out of a given `try` block falls under the former, naturally; and letting unexpected exception types bubble up counts as the latter.

Maybe the distinction could be made instead by calling a logger inside the `except` block, but simply not catching exceptions that aren't "normal" gets you a logged traceback for free. 🤷‍♂️

Reply:
Hmm, recording the exception information is a fair point, partial revert in 13b93895